samples/apps/logging/logging.cpp

```diff
@@ -287,6 +314,76 @@ namespace loggingapp
         .install();
       // SNIPPET_END: install_record
 
+      auto add_txid_in_body_put = [](auto& ctx, const auto& tx_id) {
+        static constexpr auto CCF_TX_ID = "x-ms-ccf-transaction-id";
+        ctx.rpc_ctx->set_response_header(CCF_TX_ID, tx_id.to_str());
+
+        const nlohmann::json body_j =
+          nlohmann::json::parse(ctx.rpc_ctx->get_response_body());
+
+        auto interm = body_j.get<LoggingPut::Intermediate>();
+
+        if (interm.fail)
+        {
+          throw std::runtime_error("oops, might have failed serialization");
+        }
+
+        interm.out.tx_id = tx_id.to_str();
+
+        ctx.rpc_ctx->set_response_body(nlohmann::json(interm.out).dump());
+      };
+
+      auto record_v2 = [this](auto& ctx, nlohmann::json&& params) {
+        const auto in = params.get<LoggingRecord::In>();
+
+        if (in.msg.empty())
+        {
+          return ccf::make_error(
+            HTTP_STATUS_BAD_REQUEST,
+            ccf::errors::InvalidInput,
+            "Cannot record an empty log message.");
+        }
+
+        auto records_handle =
+          ctx.tx.template rw<RecordsMap>(private_records(ctx));
+        records_handle->put(in.id, in.msg);
+        update_first_write(ctx.tx, in.id, true, get_scope(ctx));
+
+        const auto parsed_query =
+          http::parse_query(ctx.rpc_ctx->get_request_query());
+
+        std::string error_reason;
+        std::string fail;
+        if (!http::get_query_value(
+              parsed_query, "fail", fail, error_reason))
+        {
+          return ccf::make_error(
+            HTTP_STATUS_BAD_REQUEST,
+            ccf::errors::InvalidQueryParameterValue,
+            std::move(error_reason));
+        }
+
+        LoggingPut::Out resp;
+        resp.success = true;
+        LoggingPut::Intermediate interm;
+        interm.out = resp;
+        if (fail == "true") {
+          interm.fail = true;
+        }else {
```

Comment:
@jeffa5 Can you run clang-format on this code? `scripts/ci-checks.sh -f` will run all of the linters and formatters, and auto-fix what it can.